[Feat] #51 - 검색 Domain, Data 레이어 구현

yangsubinn opened the pull request:
### 🛠 작업 내용
<!-- 작업한 내용을 적어주세요. -->
- PostSearchEntity, PostSearchTransform 추가
- PostListUseCase 구현
- PostListRepository 구현, makeMockSearchResultEntity 추가
- PostVC, PostViewModel bind

### 🚨 참고 사항
<!-- 참고할 사항이 있다면 적어주세요. -->
- 원래는 PostSearch와 PostList가 동일한 entity 형태로 들어올거라 생각해서 PostListEntity를 건들였다가, api 명세서를 보고 따로 생성했슴니다(그래서 PostListEntity 등등을 좀 건들였습니다..)
- PostListVC의 테이블뷰 셀 구조가 동일해서 PostListModel은 동일하게 쓸 수 있을거라 생각하고 별도로 생성하지 않았습니다
- 지금은 mock 데이터를 넣어뒀는데, 실시간 검색이라서 네트워크 연결하면서 제대로 작동되는지 또 다시 확인해봐야될 것 같슴니다..
- diffableDatasource를 사용했더니 처음에 applySnapShot할때 조금 샤라락 등장하게 되는데, 뒤에 네트워크 연결하면서 첫번째에만 애니메이션을 false로 하던지 등..의 조치를 취하겠슴니다

### 📸 스크린샷
|기능|스크린샷|
|:--:|:--:|
|GIF|파일첨부|

### 💡 관련 이슈
<!-- 관련있는 이슈 번호(#000)를 적어주세요. -->
closed #51

yangsubinn committed:
[Add] #51 - PostSearchEntity, PostSearchTransform 추가

yangsubinn committed:
[Feat] #51 - Data 레이어 구현 및 repository mock 데이터 추가, PostListEntity 삭제

yangsubinn committed:
[Feat] #51 - Domain, Presentation 레이어 구현

yangsubinn committed:
[Chore] #51 - 상단 모듈명 변경

yangsubinn committed:
[Del] #51 -프린트문 삭제

L-j-h-c commented on SOPT-iOS/Projects/Data/Sources/Repository/PostListRepository.swift:
```diff
@@ -14,13 +14,30 @@ import Network
+extension PostListRepository {
+    private func makeMockSearchResultEntity() -> AnyPublisher<[PostListModel]?, Error> {
+        let mockSearchData = [
+            PostSearchData(noticeID: 1, title: "31th SOPT OT 공지", creator: "관리", createdAt: "2000.1.1"),
+            PostSearchData(noticeID: 2, title: "3@th SOPT OT 공지", creator: "관리", createdAt: "1999.1.1"),
+            PostSearchData(noticeID: 4, title: "3--th SOPT OT 공지", creator: "관리", createdAt: "10.1.1")
```
언어유희 제법이시네요..

L-j-h-c commented on SOPT-iOS/Projects/Data/Sources/Repository/PostListRepository.swift:
```diff
@@ -14,13 +14,30 @@ import Network
+        ]
+        let mockSearch = PostSearchEntity(notices: mockSearchData)
+        let model = mockSearch.toDomain()
+        return Just(model)
+                    .setFailureType(to: Error.self)
+                    .eraseToAnyPublisher()
```
👍 👍

L-j-h-c commented on SOPT-iOS/Projects/Presentation/Sources/PostListScene/VC/PostListVC.swift:
```diff
@@ -186,8 +183,24 @@ extension PostListVC {
+                print("event: \(event)")
+            }, receiveValue: { value in
+                self.searchResultList = value
+                if self.searchResultList.isEmpty {
+                    self.setSearchEmpty(true)
+                    self.searchEmptyLabel.text = I18N.Search.noSearchData
+                } else {
+                    self.setSearchEmpty(false)
+                    self.applySnapshot()
+                }
```
👍 👍

L-j-h-c reviewed (approved):
멋진 코드 감사합니다... 같은 뷰 작업하는데 별탈없이(?) 진행된것같아서 다행이에요~~!

lsj8706 commented on SOPT-iOS/Projects/Domain/Sources/Model/PostDetailModel.swift:
```diff
@@ -1,6 +1,6 @@
 //
 //  PostDetailModel.swift
-//  Presentation
+//  Domain
```
어이쿠 예전에 만든 파일들 주석 변경까지... 정말 감사합니다 ㅠㅠ

lsj8706 reviewed (approved):
정말 수고하셨습니다!!
저도 이 방식 참고해서 구현하도록 할게요!!

devxsby reviewed (approved):
우왁 빠르시네여,,..
참고해서 볼게요!! 감사합니다!!! 💜

yangsubinn committed:
[Merge] debelop pull
Merge branch 'develop' of https://github.com/sopt-makers/SOPT-iOS into feature/#51-SearchDomainData
